src/miner.cpp

```diff
@@ -153,6 +153,74 @@ CBlockTemplate* CreateNewBlock(const CChainParams& chainparams, const CScript& s
+            txNew.vout.back().scriptPubKey << OP_TRUE;
+            has_block_final = false;
+        }
+
+        // Otherwise we will need to check if the prior block-final transaction
+        // was a coinbase and if insufficient blocks have occured for it to
+        // mature.
+        if (has_block_final) {
+            // Fetch the last block-final tx. The index, 0, is just a
+            // placeholder and might not be the actual index we're looking for,
+            // but that's okay. This call should never fail because the prior
+            // block-final transaction was the last processed transaction (so
+            // none of the outputs could have been spent) or a previously
+            // immature coinbase.
+            COutPoint prevout(pcoinsTip->GetFinalTx(), 0);
+            const CCoins* prev_final = pcoinsTip->AccessCoins(prevout.hash);
+            if (!prev_final) {
+                return NULL;
+            }
+            // If it was a coinbase, meaning we're in the first 100 blocks after
+            // activation, then we need to make sure it has matured, otherwise
+            // we do nothing at all.
+            if (prev_final->IsCoinBase() && (nHeight - prev_final->nHeight < COINBASE_MATURITY)) {
+                // Still maturing. Nothing to do.
+                has_block_final = false;
+            } else {
+                // Create block-final tx
+                CMutableTransaction txFinal;
+                txFinal.nVersion = 2;
+                txFinal.nLockTime = txNew.nLockTime;
+                txFinal.lock_height = txNew.lock_height;
+                txFinal.vout.resize(1);
+                txFinal.vout[0].scriptPubKey << OP_TRUE;
+                // Add spendable outputs from the prior block-final transaction.
+                // We do nothing here to prevent selected transactions from from
+                // spending these same outputs out from underneath us; we depend
+                // insted on mempool protections that prevent such transactions
+                // from being considered in the first place.
+                for (; prevout.n < prev_final->vout.size(); ++prevout.n) {
+                    if (IsTriviallySpendable(*prev_final, prevout, MANDATORY_SCRIPT_VERIFY_FLAGS|SCRIPT_VERIFY_CLEANSTACK)) {
+                        txFinal.vin.push_back(CTxIn(prevout, CScript(), CTxIn::SEQUENCE_FINAL));
+                    }
+                }
+                // The previous loop should have found and added at least one
+                // input from the prior block-final transaction, so this check
+                // should never pass. The situation in which it would is if the
+                // fork is aborted after lock-in or activation, a truly abornmal
+                // circumstance in which this version of the client should stop
+                // generating blocks anyway.
+                if (txFinal.vin.empty()) {
+                    return NULL;
```

Comment:
Use `nullptr` instead of `NULL`.

Reply:
v12 is pre-C++x11 :(

Reply:
Not sure what you are saying. Core uses C++11 and uses `nullptr`, fwiw.